pkg/discovery/discovery.go

```diff
@@ -114,3 +120,59 @@ func (td *tidbDiscovery) Discover(advertisePeerUrl string) (string, error) {
 	delete(currentCluster.peers, podName)
 	return fmt.Sprintf("--join=%s", strings.Join(membersArr, ",")), nil
 }
+
+func (td *tidbDiscovery) DiscoverDM(advertisePeerUrl string) (string, error) {
+	td.lock.Lock()
+	defer td.lock.Unlock()
+
+	if advertisePeerUrl == "" {
+		return "", fmt.Errorf("dm advertisePeerUrl is empty")
+	}
+	klog.Infof("dm advertisePeerUrl is: %s", advertisePeerUrl)
+	strArr := strings.Split(advertisePeerUrl, ".")
+	if len(strArr) != 4 {
```

Comment:
IF we require the namespace to be the same as we have checked in L140, we can just take `pd-0.pd-peer` as the URL to decrease the DNS queries.